src/controller/java/templates/partials/encode_value.zapt

```diff
@@ -0,0 +1,58 @@
+{{#if isOptional}}
+chip::JniReferences::GetInstance().GetOptionalValue({{label}}, {{label}});
+{{/if}}
+{{#if isNullable}}
+  {{#if_chip_enum type}}
+    decltype(request.{{label}}) {{label}}Primitive;
+    if ({{label}} != nullptr) {
+      {{label}}Primitive = static_cast<decltype(request.{{label}})>(chip::JniReferences::GetInstance().{{asJavaBoxedType type}}ToPrimitive({{label}}));
+    }
+  {{else}}
+  {{#if_is_bitmap type}}
+    decltype(request.{{label}}) {{label}}Primitive;
+    if ({{label}} != nullptr) {
+      {{label}}Primitive = static_cast<decltype(request.{{label}})>(chip::JniReferences::GetInstance().{{asJavaBoxedType type}}ToPrimitive({{label}}));
+    }
+  {{else}}
+    {{chipType}} {{label}}Primitive;
```

Comment:
I might be missing something, but what happens for a nullable array or struct or string here?